src/util.ts

```diff
@@ -17,16 +17,24 @@ export function safeJsonParse(input: any, defaultValue: unknown): unknown {
   }
 }
 
-export function serializeObject(obj: unknown): object {
-  if (obj && typeof obj === 'object') {
-    return Object.getOwnPropertyNames(obj).reduce((map, key) => {
+export const redactSecret = (data: string): string => {
+  return data.replace(
+    /(\\*"*'*client_secret\\*"*'*:\s*\\*"*'*)([^"'\\]{60,})(\\*"*'*)/gi,
+    (m, p1, p2, p3) => `${p1}<REDACTED>${p3}`,
+  );
+};
+
+export function serializeObject(obj: unknown, redact?: boolean): object {
+  const modObj = !redact ? obj : JSON.parse(redactSecret(JSON.stringify(obj)));
```

Comment:
Why not call the `redactSecret` on the last line where we go through stringify -> parse cycle already?

Reply:
if it's already an object you return sooner

Reply:
- Can you describe that in a bit more detail, I don't follow what you mean.
- Also, can we avoid negative logic in the condition so that it is more natural to read? E.g. write it like `redactSecret ? redactSecret(...) : obj`
- And lastly, why do we have the ternary operator now twice in the function, that looks to be more complicated than it needs to be.

Reply:
sure: if it's an object you're returning here and won't get to the last line   
`return Object.getOwnPropertyNames(modObj).reduce((map, key) => {   
            // eslint-disable-next-line no-param-reassign   
            map[key] = modObj[key];   
            return map;   
        }, {});`   
ok will change the negative conditions   
I added the second ternary operator just to not do the stringify and parse twice

Reply:
Ok, I see now. I've missed there is also `return` when we do the `reduce` function.

Still, can we write this in a way that avoids the double condition with `redact`? It makes the first and the last line in the function coupled. I.e. the last line needs to be aware of what happened on the first line... so it creates a pit of failure for someone who changes only one line not being aware of the other coupled line.

Read about the Pit of success here https://blog.codinghorror.com/falling-into-the-pit-of-success/  ℹ️

Reply:
ok the last ternary operator

Reply:
No hack, it was just a shortcut to avoid the JSON parse/stringify if it was already an object. I think the new simplified code is good.